Interface 'TypedDOMStringList<T>' incorrectly extends interface 'DOMStringList'

<!-- Bug reports without a minimal reproducible example will be closed (https://stackoverflow.com/help/minimal-reproducible-example). Ideally, the example should be accessible somewhere public, like jsbin, Glitch, jsfiddle, codepen etc etc -->

trying to build this library with typescript version 5.0.4 I get this error:

```shell
node_modules/idb/build/entry.d.ts:121:18 - error TS2430: Interface 'TypedDOMStringList<T>' incorrectly extends interface 'DOMStringList'.
  The types returned by '[Symbol.iterator]()' are incompatible between these types.
    Type 'IterableIterator<T>' is missing the following properties from type 'ArrayIterator<string>': map, filter, take, drop, and 9 more.

121 export interface TypedDOMStringList<T extends string> extends DOMStringList {
                     ~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/idb/build/entry.d.ts:121

```

After some investigations I found out in the newer versions of javascript the type for iteration on DOMStringList has been officially added.

https://github.com/microsoft/TypeScript/blob/main/src/lib/dom.iterable.generated.d.ts#L80-L82

```typescript
interface DOMStringList {
    [Symbol.iterator](): ArrayIterator<string>;
}
```

now updating 

```typescript
export interface TypedDOMStringList<T extends string> extends DOMStringList {
  contains(string: T): boolean;
  item(index: number): T | null;
  [index: number]: T;
  [Symbol.iterator](): IterableIterator<T>;
}
```

to 

```typescript
export interface TypedDOMStringList<T extends string> extends DOMStringList {
  contains(string: T): boolean;
  item(index: number): T | null;
  [index: number]: T;
  [Symbol.iterator](): ArrayIterator<string>;
}
```

breaks your library because typescript version 4.8.3 and it does not have `ArrayIterator` type.

Trying to fix it I found out that removing this line:

https://github.com/jakearchibald/idb/blob/main/src/entry.ts#L261

```typescript
  [Symbol.iterator](): IterableIterator<T>;
```

does not break your library and i'm still able to build it successfully, 

Could you investigate this? if it's possible I can open a PR to remove that line.

--- a/src/entry.ts
+++ b/src/entry.ts
@@ -251,6 +251,8 @@ type IDBPDatabaseExtends = Omit<
   'createObjectStore' | 'deleteObjectStore' | 'transaction' | 'objectStoreNames'
 >;
 
+export type DOMStringListSymbolIteratorType = DOMStringList extends { [Symbol.iterator](): infer R } ? R : IterableIterator<string>;
+
 /**
  * A variation of DOMStringList with precise string types
  */
@@ -258,7 +260,12 @@ export interface TypedDOMStringList<T extends string> extends DOMStringList {
   contains(string: T): boolean;
   item(index: number): T | null;
   [index: number]: T;
-  [Symbol.iterator](): IterableIterator<T>;
+
+  /**
+   * To resolve https://github.com/jakearchibald/idb/issues/327,
+   * and for compatibility with TypeScript >= 5.6 with ArrayIterator.
+   */
+  [Symbol.iterator](): IterableIterator<string> extends DOMStringListSymbolIteratorType ? IterableIterator<T> : DOMStringListSymbolIteratorType & Iterator<T>;
 }
 
 interface IDBTransactionOptions {

